Stock_Signaller_App.py

```diff
@@ -0,0 +1,112 @@
+# Import packages
+# import os
+import requests
+import time
+# from dotenv import load_dotenv
+
+# Load environment variables from .env file 
+# load_dotenv()
+
+# API_KEY = os.environ['API_KEY']
+# PRICE_DROP_KEY = os.environ['PRICE_DROP_KEY']
+# average_drop_key = os.environ['AVERAGE_DROP_KEY']
+
+# Replace with your IFTTT Webhooks keys and event names
+price_drop_key = 'hk1XqN9c6Lf3fny61cgEOLQb3L_J71xc5h2WpNu9MmP'
+price_drop_event = 'sp_25p_alert' 
+# average_drop_key="hk1XqN9c6Lf3fny61cgEOMkLhEGT8CNupRK-3l5KgWy"
+average_drop_event = 'sp_7da_alert'
+
+# Define a list of stock symbols you want to monitor
+stock_symbols = ["TSLA", "AAPL", "GOOGL", "MSFT", "NKE"]
+
+# Dictionaries to store data
+last_prices = {}
+seven_day_averages = {}
+
+# Define the max number of API calls before stopping
+max_api_calls = 1  # Change this to your desired limit
+
+# Counter to keep track of API calls
+api_call_count = 0
+
+def send_webhook_alert(event_key, event_name, message):
+    # Send a webhook alert
+    url = f'https://maker.ifttt.com/trigger/{event_name}/with/key/hk1XqN9c6Lf3fny61cgEOMkLhEGT8CNupRK-3l5KgWy'
+    playload = {'value1': message}
+    response = requests.post(url, data=playload)
+    if response.status_code == 200:
+        print(f"Alert sent successfully to {event_name}.")
+    else:
+        print(f"Failed to send alert to {event_name}. Status code: {response.status_code}")
+
+def fetch_stock_data(symbol):
```

Comment:
This function has multiple responsibilities. Would it make sense to split it into smaller functions for better readability and maintainability?